tests/pytorch/test_sanity.py

```diff
@@ -47,6 +47,7 @@
 from transformer_engine.pytorch.tensor.utils import replace_raw_data
 from transformer_engine.pytorch.distributed import checkpoint
 from test_numerics import reset_rng_states, dtype_tols
+from fused_attn.test_fused_attn import ModelConfig, _get_attention_backends
```

Comment:
`test_fused_attn.py` is kind of a strange place to keep `ModelConfig`. It's indicating that these tests are becoming over-tuned to attention. We could avoid this refactor by moving `test_sanity_attention_extra_state` into the fused attention tests:
https://github.com/NVIDIA/TransformerEngine/blob/7b249ae82d92432936619fbe51f119c5f148a89c/tests/pytorch/test_sanity.py#L1119

Alternatively, we could move `ModelConfig` from `test_fused_attn.py` into `utils.py`. It has many attention-specific options though, so I'd prefer having separate implementations.

Reply:
There are many attention-specific args in `ModelConfig`. For linear layers, the only configs that matter are the BSHD dims so we can initialize the data tensors. I still think it's better to have a basic config in `test_numerics.py` and an attention-specific config in `test_attention.py`. At the least, `ModelConfig` should have defaults so `test_numerics.py` doesn't need much attention-specific knowledge.